pkgs/development/tools/yourkit-java/default.nix

```diff
@@ -0,0 +1,70 @@
+{ fetchzip
+, lib
+, stdenv
+, copyDesktopItems
+, makeDesktopItem
+, jre
+}:
+let
+  vPath = v: lib.elemAt (lib.splitString "-" v) 0;
+
+  arch = let inherit (stdenv.targetPlatform) system;
+         in if system == "x86_64-linux" then "x64"
+            else if system == "aarch64=linux" then "arm64"
+            else throw "Unsupported system";
+
+  desktopItem = makeDesktopItem {
+    name = "Yourkit JavaProfiler";
+    exec = "yourkit-java-profiler";
+    comment = "Award winning, fully featured low overhead profiler for Java EE and Java SE platforms";
+    desktopName = "YourKit JavaProfiler";
+    type = "Application";
+    icon = "yourkit-java-profiler";
+    terminal = false;
+    categories = [ "Development" ];
+    startupWMClass = "yourkit-java-profiler";
+    startupNotify = false;
+  };
```

Comment:
I just noticed that when I run the application, it drops a `yourkit-yjp.desktop` into `~/.local/share/applications`, which I suspect will lead to a duplicate entry with this. So it may not in fact be a good idea to package a desktop file after all, sorry about that 😓

Reply:
Unless it can be convinced not to create that desktop file, of course.

Reply:
> I just noticed that when I run the application, it drops a `yourkit-yjp.desktop` into `~/.local/share/applications`, which I suspect will lead to a duplicate entry with this. So it may not in fact be a good idea to package a desktop file after all, sorry about that 😓

I prefer our desktop item, which gets cleaned when the package is removed. I have installed a patch to forbid the profiler to create such an entry itself.